### DISCUSSION TITLE
Question about the dynamics computation library

### DISCUSSION THREAD

**Discussion by makolon** [haosulab/SAPIEN#159] [Q&A]
Hi! Thank you for sharing such a great physics simulator. I've been using SAPIEN through ManiSkill!

I have a question about the dynamics computation library in the SAPIEN simulator. I understand that SAPIEN uses PhysX as its underlying physics engine, and that forward and inverse dynamics are also implemented using PhysX. However, I noticed that some calculations can be done using Pinocchio as well. Could you please explain the reason for providing the option to calculate dynamics using Pinocchio?

Thank you!

**Comment by StoneT2000** [comment]
Pinnochio is a pretty popular library, we primarily use it for inverse kinematics in ManiSkill as far as I am aware, and we also use a library from our lab called fast_kinematics for inverse kinematics (jacobian computation) for GPU simulation.

**Reply by makolon** [reply]
Thank you!
